### PR TITLE
Electra core transition operations

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**

taken from the [electra-devnet0](https://github.com/prysmaticlabs/prysm/tree/electra-devnet0) branch, and adds some place holders for Electra core transition. subsequent PRs will address the contents of the processing as well as add tests

spec changes can be found https://github.com/ethereum/consensus-specs/tree/dev/specs/electra

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/core/transition/transition_no_verify_sig.go [hunk 1]
```diff
@@ -378,6 +394,72 @@ func VerifyBlobCommitmentCount(blk interfaces.ReadOnlyBeaconBlock) error {
...
+	st, err = electra.ProcessExecutionLayerWithdrawRequests(ctx, st, exe.WithdrawalRequests())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not process execution layer withdrawal requests")
+	}
+	st, err = electra.ProcessDepositReceipts(ctx, st, exe.DepositReceipts())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not process deposit receipts")
+	}
+
+	if err := electra.ProcessConsolidations(ctx, st, bod.Consolidations()); err != nil {
```
Comment (james-prysm) [inline beacon-chain/core/transition/transition_no_verify_sig.go:457]: why does this not return a state like the other process functions?

Comment (prestonvanloon) [reply beacon-chain/core/transition/transition_no_verify_sig.go:457]: @rkapka made me do it :rofl: 

The better question is why do the other methods return the state when they are mutating the state argument :thinking:

File: testing/spectest/minimal/electra/fork_transition/transition_test.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package fork_transition
+
+import (
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/v5/testing/spectest/shared/electra/fork"
+)
+
+func TestMainnet_Electra_Transition(t *testing.T) {
```
Comment (prestonvanloon) [inline testing/spectest/minimal/electra/fork_transition/transition_test.go:9]:
```suggestion
func TestMinimal_Electra_Transition(t *testing.T) {
```

File: testing/spectest/minimal/electra/fork_transition/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,7 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    srcs = ["transition_test.go"],
+    deps = ["//testing/spectest/shared/electra/fork:go_default_library"],
+)
```
Comment (prestonvanloon) [inline testing/spectest/minimal/electra/fork_transition/BUILD.bazel:7]:
```suggestion
go_test(
    name = "go_default_test",
    srcs = ["transition_test.go"],
    deps = ["//testing/spectest/shared/electra/fork:go_default_library"],
    network = "minimal",
    data = glob(["*.yaml"]) + [
        "@consensus_spec_tests_minimal//:test_data",
    ],
)
```

Run gazelle too, i dont think i put the fields in the right order

File: testing/spectest/mainnet/electra/fork_transition/transition_test.go [hunk 4]
```diff
@@ -7,6 +7,5 @@ import (
 )
 
 func TestMainnet_Electra_Transition(t *testing.T) {
-	t.Skip("TODO: Electra")
```
[the remark this answers is not in the document]
Comment (james-prysm) [reply testing/spectest/mainnet/electra/fork_transition/transition_test.go]: no i forgot to skip it

File: beacon-chain/core/electra/effective_balance_updates.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+//	            else MIN_ACTIVATION_BALANCE
+//	        )
+//
+//	        if (
+//	            balance + DOWNWARD_THRESHOLD < validator.effective_balance
+//	            or validator.effective_balance + UPWARD_THRESHOLD < balance
+//	        ):
+//	            validator.effective_balance = min(balance - balance % EFFECTIVE_BALANCE_INCREMENT, EFFECTIVE_BALANCE_LIMIT)
+func ProcessEffectiveBalanceUpdates(state state.BeaconState) error {
+	// TODO: replace with real implementation
```
Comment (prestonvanloon) [inline beacon-chain/core/electra/effective_balance_updates.go:27]: Coming in #14003

File: beacon-chain/core/electra/registry_updates.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+//		        ):
+//		            initiate_validator_exit(state, ValidatorIndex(index))
+//
+//	         # Activate all eligible validators
+//	         activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
+//	         for validator in state.validators:
+//	             if is_eligible_for_activation(state, validator):
+//	                 validator.activation_epoch = activation_epoch
+func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) (state.BeaconState, error) {
+	// TODO: replace with real implementation
```
Comment (prestonvanloon) [inline beacon-chain/core/electra/registry_updates.go:32]: Coming in #14005

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**